src/Framework/NativeMethods.cs

```diff
@@ -435,72 +450,85 @@ public SystemInformationData()
                     Architecture.X86 => ProcessorArchitectures.X86,
+#if NET5_0_OR_GREATER
+                    Architecture.Wasm => ProcessorArchitectures.WASM,
+#endif
+#if NET6_0_OR_GREATER
+                    Architecture.S390x => ProcessorArchitectures.S390X,
+#endif
+#if NET7_0_OR_GREATER
+                    Architecture.LoongArch64 => ProcessorArchitectures.LOONGARCH64,
+                    Architecture.Armv6 => ProcessorArchitectures.ARMV6,
+                    Architecture.Ppc64le => ProcessorArchitectures.PPC64LE,
+#endif
                     _ => ProcessorArchitectures.Unknown,
                 };
-#endif
-                // Fall back to 'uname -m' to get the architecture.
-                if (processorArchitecture == ProcessorArchitectures.Unknown)
+
+#else
+                // Mono
+                // Use 'uname -m' to get the architecture.
+                try
                 {
-                    try
+                    // On Unix run 'uname -m' to get the architecture. It's common for Linux and Mac
+                    using (
+                        var proc =
+                            Process.Start(
+                                new ProcessStartInfo("uname")
+                                {
+                                    Arguments = "-m",
+                                    UseShellExecute = false,
+                                    RedirectStandardOutput = true,
+                                    CreateNoWindow = true
+                                }))
                     {
-                        // On Unix run 'uname -m' to get the architecture. It's common for Linux and Mac
-                        using (
-                            var proc =
-                                Process.Start(
-                                    new ProcessStartInfo("uname")
-                                    {
-                                        Arguments = "-m",
-                                        UseShellExecute = false,
-                                        RedirectStandardOutput = true,
-                                        CreateNoWindow = true
-                                    }))
+                        string arch = null;
+                        if (proc != null)
+                        {
+                            arch = proc.StandardOutput.ReadLine();
+                            proc.WaitForExit();
+                        }
+
+                        if (!string.IsNullOrEmpty(arch))
                         {
-                            string arch = null;
-                            if (proc != null)
+                            if (arch.StartsWith("x86_64", StringComparison.OrdinalIgnoreCase))
                             {
-                                arch = proc.StandardOutput.ReadLine();
-                                proc.WaitForExit();
+                                processorArchitecture = ProcessorArchitectures.X64;
```

Comment:
These cases should have been assigning `processorArchitecture`.

It's a regression that happened when I added the runtime architecture detection in https://github.com/dotnet/msbuild/pull/5897.